chromadb/segment/impl/manager/local.py

```diff
@@ -130,16 +141,30 @@ def delete_segments(self, collection_id: UUID) -> Sequence[UUID]:
                     instance = self.get_segment(collection_id, MetadataReader)
                     instance.delete()
                 del self._instances[segment["id"]]
-            if collection_id in self._segment_cache:
-                if segment["scope"] in self._segment_cache[collection_id]:
-                    del self._segment_cache[collection_id][segment["scope"]]
-                del self._segment_cache[collection_id]
+            if segment["scope"] is SegmentScope.VECTOR:
+                self.segment_cache[SegmentScope.VECTOR].pop(collection_id)
+            if segment["scope"] is SegmentScope.METADATA:
+                self.segment_cache[SegmentScope.METADATA].pop(collection_id)
         return [s["id"] for s in segments]
 
     @trace_method(
         "LocalSegmentManager.get_segment",
         OpenTelemetryGranularity.OPERATION_AND_SEGMENT,
     )
+    def _get_segment_disk_size(self, collection_id: UUID) -> int:
+        segments = self._sysdb.get_segments(collection=collection_id, scope=SegmentScope.VECTOR)
+        if len(segments) == 0:
+            return 0
+        size = get_directory_size(
+            os.path.join(self._system.settings.require("persist_directory"), str(segments[0]["id"])))
```

Comment:
nit: leave comment stating assumption of one vector segment for this line - otherwise hardcoded [0] is confusing